sui-execution/latest/sui-verifier/src/entry_points_verifier.rs

```diff
@@ -234,6 +235,16 @@ fn verify_param_type(
         ));
     }
 
+    // Only `sui::sui_system` is allowed to expose entry functions that accept a mutable Random
+    // parameter.
+    if is_mutable_random(view, param) {
```

Comment:
Random is already landed in testnet/mainnet right? 

If so, you will have to gate this via a protocol config. 

And since you will have to protocol config this (I presume)... 
We should probably just extend this to _all_ functions (excluding test functions). 
So with this protocol config, we should ban `&mut Clock` and `&mut Random` across the board.

Happy to help with the compiler checks for that too